### PR TITLE
Simplify handling of `LocalDate` value with invalid trailing time-portion

### PR DESCRIPTION
**Pull request by cowtowncoder**
As per title: test change just cleaning up, but for `LocalDate`, avoid unnecessary handling of timezone.

### CODE REVIEW

File: datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/LocalDateDeserializer.java [hunk 1]
```diff
@@ -155,7 +155,8 @@ protected LocalDate _fromString(JsonParser p, DeserializationContext ctxt,
                 // JavaScript by default includes time in JSON serialized Dates (UTC/ISO instant format).
                 if (string.length() > 10 && string.charAt(10) == 'T') {
                    if (string.endsWith("Z")) {
-                       return LocalDateTime.ofInstant(Instant.parse(string), ZoneOffset.UTC).toLocalDate();
+                       return LocalDate.parse(string.substring(0, string.length() - 1),
+                               DateTimeFormatter.ISO_LOCAL_DATE_TIME);
```
Comment (cowtowncoder) [inline datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/LocalDateDeserializer.java:159]: Seems to me that there is no need to go back and forth to zoned instant when simply trying to retain local value.
Tests all pass as well.

**Comment by cowtowncoder** [comment]
@kupci Just noticed this when looking at LocalDate/LocalDateTime deserialization; wanted to sanity check my thinking here.

**Review by kupci — APPROVED** [review]
> Should not contain time component when 'strict' mode set for property or type (enable 'lenient' handling to allow)

Nice solution! I see this question coming up every so often, so this looks like a good improvement.

**Comment by cowtowncoder** [comment]
I copied that from `LocalDateTime` deserializer where it already existed. I hope logic makes sense; it should as lenient is the default.

I actually should file an issue for that change, and simplification can just be part of implementation for that issue.

**Pull request merged**